src/Compilers/CSharp/Portable/Symbols/Metadata/PE/ExternalAnnotations.cs

```diff
@@ -0,0 +1,98 @@
+                { "System.Void System.Buffer.SetByte(System.Array, System.Int32, System.Byte)", Parameters(skip, Nullable(false), skip, skip) },
+                { "System.Byte System.Byte.Parse(System.String)", Parameters(skip, Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.IFormatProvider)", Parameters(skip, Nullable(false), skip) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles)", Parameters(skip, Nullable(false), Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles, System.IFormatProvider)", Parameters(skip, Nullable(false), Nullable(false), skip) },
+                { "System.String System.String.Concat(System.String, System.String)", Parameters(Nullable(false), Nullable(true), Nullable(true)) },
+            }.ToImmutableDictionary();
+
+        internal static string MakeMethodKey(PEMethodSymbol method, ParamInfo<TypeSymbol>[] paramInfo)
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+
+            StringBuilder builder = pooledBuilder.Builder;
+            Add(paramInfo[0].Type, builder);
+            builder.Append(' ');
+
+            Add(method.ContainingType, builder);
+            builder.Append('.');
+
+            builder.Append(method.Name);
+            builder.Append('(');
+
+            for (int i = 1; i < paramInfo.Length; i++)
+            {
+                Add(paramInfo[i].Type, builder);
+                if (i < paramInfo.Length - 1)
+                {
+                    builder.Append(", ");
+                }
+            }
+
+            builder.Append(')');
+            return pooledBuilder.ToStringAndFree();
+
+            // PROTOTYPE(NullableReferenceTypes): Many cases are not yet handled
+            // generic type args
+            // ref kind
+            // 'this'
+            // static vs. instance
+            // use assembly qualified name format (used in metadata) rather than symbol display?
+        }
+
+        /// <summary>
+        /// All types in a member which can be annotated should be annotated. Value types can be skipped.
+        /// </summary>
+        private static readonly ImmutableArray<bool> skip = ImmutableArray<bool>.Empty;
+
+        private static ImmutableArray<ImmutableArray<bool>> Parameters(params ImmutableArray<bool>[] values)
+            => values.ToImmutableArray();
+
+        private static ImmutableArray<bool> Nullable(params bool[] values)
+            => values.ToImmutableArray();
+
+        internal static ImmutableArray<ImmutableArray<bool>> GetExtraAnnotations(string key)
+        {
+            if (!Annotations.TryGetValue(key, out var flags))
+            {
+                return default;
+            }
+
+            return flags;
+        }
+
+        private static void Add(TypeSymbol type, StringBuilder builder)
+            => builder.Append(
```

Comment:
:memo: Sadly, it looks like this use of symbol display is sufficient to cause an infinite regress. I'm investigating the scenario that's resulting in CI stack overflow (something to do ValueTuple used in a method on ValueTuple, but to print out a tuple we need to resolve all the members of ValueTuple, to get possible tuple element names).

Reply:
> but to print out a tuple we need to resolve all the members of ValueTuple

Why are we using an option that pretty-prints tuples here?

Reply:
@CyrusNajmabadi I can't imagine that such a loop exists today (before this PR). To hit this, you'd need to try and display a symbol in code that contributes to constructing the symbol.

Reply:
What if you're in the middle of editing ValueTuple struct source code?

Reply:
I think that is fine. You have a `ValueTuple` symbol that can successfully be constructed (with whatever members or errors it has at the time) and then we can display it.
The problem is when the constructor of the symbol object try to display the symbol.